modules/reindex/src/main/java/org/elasticsearch/index/reindex/DeleteByQueryRequest.java

```diff
@@ -76,4 +81,10 @@ public String toString() {
         searchToString(b);
         return b.toString();
     }
+
+    @Override
+    public List<? extends IndicesRequest> subRequests() {
+        assert getSearchRequest() != null;
+        return Collections.singletonList(getSearchRequest());
```

Comment:
Shield uses it as a first pass. The actual security is rechecked on the search and sub-bulk requests so this is just more of a pre-flight thing.

It makes sense to return both requests, yeah.